docs/development/apifirst.md

```diff
@@ -0,0 +1,133 @@
+---
```

Comment:
I'd suggest we change the file to be apis.md instead of apifirst.md.  I'm thinking this section should talk about experience/recommendations for creating APIs with API first being the approach many teams have had success with/would recommend.

Reply:
Would that be confusing?  Many people will see it as "build your endpoints using express guide". 
Where this is dedicated to Patterns on Building restful API.

I intentionally avoided being biased in the doc by discussing topics in context of API first - meaning that team will need to understand it and want to use it.